readme.md

```diff
@@ -1,15 +1,108 @@
+  "pdfIdentifier": "a17_flightplan",
+  "outputLocation": "s3://pdf-to-alto/a17_flightplan_alto"
+}
+```
+
+Where
+* `pdfLocation` - the URL where PDF can be downloaded from.
+* `pdfIdentifier` - unique identifier for PDF. Generated altos will be in format `f"{pdfIdentifier}-{i:04d}.xml"`,
+  where `i` is 0-based page index. If omitted a random uuid will be used.
+* `outputLocation` - s3 location where final ALTO files will be output. With or without preceding `s3://` and no
+  trailing `/`.
+
+(See [sample.json](/sample.json))
+
+The completed notification message echos back the original message with `"numberOfFiles"` property added.
+
+## Technology
+
+This is a Python script that utilises the following libraries:
+
+* [pdfalto](https://github.com/kermitt2/pdfalto) - C lib used to generate ALTO files.
+* [PyMuPDF](https://pymupdf.readthedocs.io/en/latest/) - Python lib used to query PDF object for page dimensions.
+* [lxml](https://lxml.de/) - Used to parse ALTO files and update scaled values.
+* [requests](https://docs.python-requests.org/en/latest/) - Used to download PDF files.
+
+## Configuration
+
+The following environment variables can be used to configure the app:
+
+| Env Var             | Description                                                          | Default               |
+|---------------------|----------------------------------------------------------------------|-----------------------|
+| DOWNLOAD_CHUNK_SIZE | Chunk size for downloading PDF                                       | 2048                  |
+| WORKING_FOLDER      | Local working folder for storing generated files                     | ./work                |
+| REMOVE_WORK_DIR     | Whether to clean up working dir on completion                        | True                  |
+| RESCALE_ALTO        | Whether to rescale generated ALTO to page                            | True                  |
+| MONITOR_SLEEP_SECS  | How long to sleep long polling operations if no messages received    | 30                    |
+| AWS_REGION          | AWS region being used                                                | eu-west-1             |
+| INCOMING_QUEUE      | The name of the SQS queue to monitor for incoming messages. Required |                       |
+| COMPLETED_TOPIC_ARN | The ARN of a topic to post completion notifications to               |                       |
+| LOCALSTACK          | If using LocalStack                                                  | False                 |
+| LOCALSTACK_ADDRESS  | Address for LocalStack instance                                      | http://localhost:4566 |
+
+(See [.env.dist](/.env.dist) for sample .env file)
 
 ## Running Locally
 
-There is a multi-stage Dockerfile that builds the [pdfalto](https://github.com/kermitt2/pdfalto) library and copies it to a new stage.
+There is a multi-stage Dockerfile that builds the `pdfalto` binary and copies it to a new stage.
```

Comment:
Any reason why we can't do all the manual `docker` commands here as part of the existing `docker-compose.local.yml` (also in the absence of a competing `docker-compose.yml`, I personally would probably just rename `docker-compose.local.yml` to `docker-compose.yml`).

Reply:
Yeah I can add `docker-compose.yml` which includes all of this. 

The issue I ran into was that LocalStack can be slow to start so when `pdf-to-alto` starts and tries to poll queue it doesn't exist yet. I couldn't find a 'nice' solution - any ideas?

The ideal behind `docker-compose.local.yml` is to contain dependencies required for running it local.

Reply:
> The issue I ran into was that LocalStack can be slow to start so when pdf-to-alto starts and tries to poll queue it doesn't exist yet. I couldn't find a 'nice' solution - any ideas?

If there's a way of writing a script to detect then LocalStack is "ready", then you could use the `command` parameter: https://docs.docker.com/compose/startup-order/ - then use the `depends_on` on the dependent service, e.g.:

```
depends_on:
  - local_stack
```

Reply:
> The ideal behind docker-compose.local.yml is to contain dependencies required for running it local.

I'm just being picky, by noting that the `.local.` implies to me that there should be one or more equivalent non-local ones. Feel free to ignore.

Reply:
No, it's a fair point. It's always good to be able to run `docker-compose up` and just get everything going.

Reply:
Added `docker-compose.yml` file, with a `wait-for-localstack.sh` startup file. Feels a bit janky but it's working.